v8env_test/request.spec.js

```diff
@@ -5,10 +5,10 @@ describe("Request", () => {
     expect(new Request("http://example.com")).not.to.throw
   })
 
-  it("sets body properly from intializing request", ()=>{
+  it("sets body properly from intializing request", () => {
     const r = new Request("https://example.com", { body: "ahoyhoy", method: "post"})
     const req = new Request(r)
 
     expect(req.bodySource).to.eq(r.bodySource)
   })
-})
\ No newline at end of file
+})
```

Comment:
I think you could test that a `new Response`, provided a `Uint8Array` as the first argument (its body), which has its method `arrayBuffer` called on it will return the same bytes that were originally given to it.

Something like:

```javascript
it('returns an ArrayBuffer given a Uint8Array', async function(){
  const bodyText = "hello world"
  const r = new Response(new TextEncoder().encode(bodyText))
  expect(new TextDecoder('utf-8').decode(await r.arrayBuffer())).to.equal(bodyText)
})
```

(untested)

Reply:
I will add that (and test it).

Reply:
I went to add my test and it looks like there is some sort of issue with the tests - I am not exactly sure what it is saying.

### Info
* This does not happen on fly/master
* This happened before I added the test you suggested

<details>
<summary>Error</summary>
<br>

```

> @fly/fly@0.27.0 test /Users/zoe/Developer/fly
> npm run test:v8env && npm run test:core


> @fly/fly@0.27.0 test:v8env /Users/zoe/Developer/fly
> npm run build:v8env && node ./lib/cmd/index.js test v8env_test/**/*.spec.js


> @fly/fly@0.27.0 build:v8env /Users/zoe/Developer/fly
> webpack --config webpack.v8.config.js

Hash: 3bb892ffce6bc1450f8b6d5bf813e3c89447074c
Version: webpack 3.11.0
Time: 2265ms
                      Asset       Size  Chunks                    Chunk Names
                   v8env.js     736 kB       0  [emitted]  [big]  main
             v8env.map.json     910 kB       0  [emitted]  [big]  main
        v8env/lib/blob.d.ts  383 bytes          [emitted]
  v8env/lib/body_mixin.d.ts  509 bytes          [emitted]
      v8env/lib/crypto.d.ts  449 bytes          [emitted]
   v8env/lib/form_data.d.ts  515 bytes          [emitted]
      v8env/lib/logger.d.ts  183 bytes          [emitted]
   v8env/lib/fly/image.d.ts    3.44 kB          [emitted]
v8env/lib/utils/buffer.d.ts   75 bytes          [emitted]
  [39] ./v8env/index.js 4.13 kB {0} [built]
  [98] ./v8env/ts/body_mixin.ts 4.35 kB {0} [built]
 [101] ./v8env/ts/blob.ts 2.21 kB {0} [built]
 [102] ./v8env/ts/form_data.ts 1.89 kB {0} [built]
 [103] ./v8env/ts/crypto.ts 882 bytes {0} [built]
 [105] ./v8env/response.js 2.29 kB {0} [built]
 [107] ./v8env/cache.js 2.55 kB {0} [built]
 [109] ./v8env/timers.js 752 bytes {0} [built]
 [110] ./v8env/utils/error.js 5.53 kB {0} [built]
 [111] ./v8env/middleware/fly-backend.js 448 bytes {0} [built]
 [112] ./v8env/middleware/fly-echo.js 751 bytes {0} [built]
 [113] ./v8env/middleware/fly-routes.js 2.78 kB {0} [built]
 [114] ./v8env/middleware/force-ssl.js 536 bytes {0} [built]
 [115] ./v8env/middleware/google-analytics.js 5.47 kB {0} [built]
 [116] ./v8env/middleware/session.js 578 bytes {0} [built]
    + 103 hidden modules
Using Webpack config ./webpack.fly.config.js
Compiling app w/ options: { watch: false }
Loaded v8 env bundle (hash: 169c508599b9a1e3d423fad9af29952b07f56db7)
Compiled app bundle (hash: 42b0e2269f819794c7f799e3fb5706e8b208fc97)
Running tests...


  crypto
    1) creates a hash
    2) creates a hash from a string
    3) creates a hash synchronously
    4) creates a hash synchronously from a string
    5) produces a string with encoding
    6) produces a string with encoding synchronously
    ✓ errors on bad algo
2018-04-05T21:51:49.325Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.326Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.326Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.326Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.333Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.334Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.334Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.334Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.335Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
2018-04-05T21:51:49.335Z - error: error getting original source position Error: Attempted to call a unregistered bridge function.
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at (<isolated-vm boundary>)
    at dispatchSync (dist/v8env.js:8096:23)
    at originalPositionFor (dist/v8env.js:21895:23)
    at mapSourcePosition (dist/v8env.js:21948:12)
    at wrapCallSite (dist/v8env.js:21919:20)
    at dist/v8env.js:21902:26
    7) errors on bad algo (sync)

  Document
    ✓ exists
    ✓ parses html into a queryable dom tree
    ✓ can stringify a DOM
    ✓ can replace elements

  FormData
    ✓ exists
    ✓ can be instantiated
    ✓ can get/set params
    ✓ can delete params
    ✓ can getAll/set/append params
    ✓ gracefully handle non-existant keys
    ✓ keys() are iterable
    ✓ values() are iterable
    ✓ entries() are iterable
    ✓ has()
    toString()
      ✓ returns empty string
      ✓ stringifies simple params
      ✓ escapes special symbols

  Image
    ✓ metadata()
    ✓ resize()
    ✓ withoutEnlargement().resize()
    ✓ crop() (97ms)
    ✓ errors with bad ops
    ✓ errors with bad data

  Request
    ✓ can be instantiated
    ✓ sets body properly from intializing request

  TextDecoder
    ✓ can be instantiated
    8) decodes

  TextEncoder
    ✓ can be instantiated
    9) encodes

  timers
    setTimeout
      ✓ calls after the right amount of time (103ms)


  29 passing (361ms)
  9 failing

  1) crypto
       creates a hash:
     Error: Attempted to call a unregistered bridge function.
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at (<isolated-vm boundary>)
      at dispatchSync (dist/v8env.js:8096:23)
      at textEncode (dist/v8env.js:20220:23)
      at encode (dist/v8env.js:20199:12)
      at it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:1756:79)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12

  2) crypto
       creates a hash from a string:
     Error: Attempted to call a unregistered bridge function.
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at (<isolated-vm boundary>)
      at dispatch (dist/v8env.js:8085:23)
      at digestHash (dist/v8env.js:20814:27)
      at digest (dist/v8env.js:20824:20)
      at it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:1761:36)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12

  3) crypto
       creates a hash synchronously:
     Error: Attempted to call a unregistered bridge function.
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at (<isolated-vm boundary>)
      at dispatchSync (dist/v8env.js:8096:23)
      at textEncode (dist/v8env.js:20220:23)
      at encode (dist/v8env.js:20199:12)
      at it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:1766:77)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12

  4) crypto
       creates a hash synchronously from a string:
     Error: Attempted to call a unregistered bridge function.
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at (<isolated-vm boundary>)
      at dispatchSync (dist/v8env.js:8096:23)
      at digestHashSync (dist/v8env.js:20817:27)
      at digestSync (dist/v8env.js:20827:20)
      at it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:1771:30)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12

  5) crypto
       produces a string with encoding:
     Error: Attempted to call a unregistered bridge function.
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at (<isolated-vm boundary>)
      at dispatch (dist/v8env.js:8085:23)
      at digestHash (dist/v8env.js:20814:27)
      at digest (dist/v8env.js:20824:20)
      at it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:1776:36)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12

  6) crypto
       produces a string with encoding synchronously:
     Error: Attempted to call a unregistered bridge function.
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at (<isolated-vm boundary>)
      at dispatchSync (dist/v8env.js:8096:23)
      at digestHashSync (dist/v8env.js:20817:27)
      at digestSync (dist/v8env.js:20827:20)
      at it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:1781:30)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12

  7) crypto
       errors on bad algo (sync):

      AssertionError: expected [Function] to throw error including 'Digest method not supported' but got 'Attempted to call a unregistered bridge function.'
      + expected - actual

      -Attempted to call a unregistered bridge function.
      +Digest method not supported

      at Context.it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:1795:123)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Test.Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12
      at next (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4667:14)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4677:7
      at next (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4601:14)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4645:5
      at timeslice (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:82:27)

  8) TextDecoder
       decodes:
     Error: Attempted to call a unregistered bridge function.
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at (<isolated-vm boundary>)
      at dispatchSync (dist/v8env.js:8096:23)
      at textDecode (dist/v8env.js:20216:23)
      at decode (dist/v8env.js:20208:12)
      at it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:13391:61)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12

  9) TextEncoder
       encodes:
     Error: Attempted to call a unregistered bridge function.
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/bridge/bridge.ts:14:27)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at Object.<anonymous> (/Users/zoe/Developer/fly/src/cmd/test.ts:9:1)
      at Module._compile (module.js:652:30)
      at Object.Module._extensions..js (module.js:663:10)
      at Module.load (module.js:565:32)
      at tryModuleLoad (module.js:505:12)
      at Function.Module._load (module.js:497:3)
      at Module.require (module.js:596:17)
      at require (internal/module.js:11:18)
      at (<isolated-vm boundary>)
      at dispatchSync (dist/v8env.js:8096:23)
      at textEncode (dist/v8env.js:20220:23)
      at encode (dist/v8env.js:20199:12)
      at it (bundle-42b0e2269f819794c7f799e3fb5706e8b208fc97.js:13402:61)
      at callFn (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4266:21)
      at Runnable.run (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4258:7)
      at Runner.runTest (/Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4753:10)
      at /Users/zoe/Developer/fly/node_modules/mocha/mocha.js:4871:12



npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @fly/fly@0.27.0 test:v8env: `npm run build:v8env && node ./lib/cmd/index.js test v8env_test/**/*.spec.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @fly/fly@0.27.0 test:v8env script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/zoe/.npm/_logs/2018-04-05T21_51_49_697Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @fly/fly@0.27.0 test: `npm run test:v8env && npm run test:core`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @fly/fly@0.27.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/zoe/.npm/_logs/2018-04-05T21_51_49_724Z-debug.log
```
</details>

Reply:
You'll have to recompile the typescript with `npm run build`.

You can also run `npm run build:live` and it'll watch for changes and automatically recompile.

Reply:
That's great - I thought I really broke something there for a moment 😆